org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/TypeUtils.java

```diff
@@ -0,0 +1,222 @@
+/*******************************************************************************
+* Copyright (c) 2017 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter;
+
+import com.sun.jdi.ClassLoaderReference;
+import com.sun.jdi.ReferenceType;
+import com.sun.jdi.Type;
+import com.sun.jdi.VirtualMachine;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+
+
+/**
+ * The type utility for retrieving display name of the type.
+ */
+public final class TypeUtils {
+    static final char ARRAY = '[';
+    static final char BYTE = 'B';
+    static final char CHAR = 'C';
+    static final char OBJECT = 'L';
+    static final char FLOAT = 'F';
+    static final char DOUBLE = 'D';
+    static final char INT = 'I';
+    static final char LONG = 'J';
+    static final char SHORT = 'S';
+    static final char VOID = 'V';
+    static final char BOOLEAN = 'Z';
+    static final char STRING = 's';
+    static final char THREAD = 't';
+    static final char THREAD_GROUP = 'g';
+    static final char CLASS_LOADER = 'l';
+    static final char CLASS_OBJECT = 'c';
+    static final char SIGNATURE_ENDCLASS = ';';
+    static final char GENERIC_START = '<';
+    static final char GENERIC_END = '>';
+    static final char COLON = ':';
+
+    static Map<Character, String> primitiveTypeMap = new HashMap<Character, String>();
+
+    static {
+        primitiveTypeMap.put(BYTE, "byte");
+        primitiveTypeMap.put(CHAR, "char");
+        primitiveTypeMap.put(FLOAT, "float");
+        primitiveTypeMap.put(DOUBLE, "double");
+        primitiveTypeMap.put(INT, "int");
+        primitiveTypeMap.put(LONG, "long");
+        primitiveTypeMap.put(SHORT, "short");
+        primitiveTypeMap.put(VOID, "void");
+        primitiveTypeMap.put(BOOLEAN, "boolean");
+    }
+
+    /**
+     * Returns the short name and signature of the JDI type, eg: <code>HashMap&lt;K,V&gt;</code> for
+     * <code>java.util.HashMap</code>.
+     *
+     * @param type
+     *            the JDI type for evaluate
+     * @return the display name
+     */
+    public static String getDisplayName(Type type) {
```

Comment:
As we discussed before, we will use the name() signature() provided by JDI. Let's focus on other parts like showing the value itself other than formatting the display name.